homeassistant/components/group/entity.py

```diff
@@ -264,11 +264,10 @@ def assumed_state(self) -> bool:
 
     def update_tracked_entity_ids(self, entity_ids: Collection[str] | None) -> None:
         """Update the member entity IDs."""
-        asyncio.run_coroutine_threadsafe(
-            self.async_update_tracked_entity_ids(entity_ids), self.hass.loop
```

Comment:
If we need to wait for the result, we still needs a coro here even if it's a protected method that calls the callback function


If we don't need to wait we can use call_soon_threadsafe instead

Reply:
It seems to me we do not need to wait here, but I am not sure where this is used.

Reply:
It never used AFAICT. Its old pre-async code everything code.

So call_soon_threadsafe seems fine.

We could probably rip it out as well in a future PR

Reply:
https://github.com/home-assistant/core/pull/119798